### PR TITLE
Fix: Correctly abort process on timeout

### PR DESCRIPTION
**Pull request by MarcoPolo**

### CODE REVIEW

File: multidim-interop/src/compose-runner.ts [hunk 1]
```diff
@@ -43,14 +44,11 @@ export async function run(namespace: string, compose: ComposeSpecification, opts
...
-        let timeoutId
-        const { stdout, stderr } =
-            (await Promise.race([
-                exec(`docker compose -f ${path.join(dir, "compose.yaml")} up ${upFlags.join(" ")}`),
-                // Timeout - uses any type because this will only reject the promise.
-                new Promise<any>((resolve, reject) => { timeoutId = setTimeout(() => reject("Timeout"), 1000 * timeoutSecs) })
-            ]))
+        const timeoutId = setTimeout(() => controller.abort(), 1000 * timeoutSecs)
+
+        const controller = new AbortController();
```
Comment (thomaseizinger) [inline multidim-interop/src/compose-runner.ts:49]: Isn't `controller` undefined in the closure? Even if this works, can we declare `controller` first please?

File: multidim-interop/src/compose-runner.ts [hunk 2]
```diff
@@ -43,14 +44,11 @@ export async function run(namespace: string, compose: ComposeSpecification, opts
...
+        const { signal } = controller;
+        const { stdout, stderr } = await exec(`docker compose -f ${path.join(dir, "compose.yaml")} up ${upFlags.join(" ")}`, { signal })
```
Comment (thomaseizinger) [inline multidim-interop/src/compose-runner.ts:51]: If `exec` needs an object that has a `signal` property, can't you just pass `controller` directly?

**Review by thomaseizinger — COMMENTED** [review]
Thanks for splitting this out into a separate PR.

File: multidim-interop/src/compose-runner.ts (hunk 2 above)
Comment (MarcoPolo) [reply multidim-interop/src/compose-runner.ts:50]: This avoids passing in other things that are in the controller. This is also how you do it per the docs: https://nodejs.org/api/child_process.html#child_processexeccommand-options-callback

File: multidim-interop/src/compose-runner.ts (hunk 1 above)
Comment (MarcoPolo) [reply multidim-interop/src/compose-runner.ts]: It's tricky, but L49 will always happen before the closure. I'll update this anyways.

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**